bundles/mapping/mapmodule/mapmodule.olcs.js

```diff
@@ -524,21 +525,27 @@ class MapModuleOlCesium extends MapModuleOl {
     }
 
     setCameraToRotateMode () {
-        const camera = this.getCesiumScene().camera;
-        // Look at target in front at current direction
-        var distanceInMeters = 5;
-        camera.zoomIn(distanceInMeters);
-        var lookAtTarget = camera.position;
-        // Place camera to original position (x meters from look at target)
-        camera.constrainedAxis = Cesium.Cartesian3.UNIT_Z;
-        var offset = new Cesium.HeadingPitchRange(camera.heading, camera.pitch, distanceInMeters);
-        camera.lookAt(lookAtTarget, offset);
+        const scene = this.getCesiumScene();
+        const { camera } = scene;
+
+        // Get target to look at with ray from globe surface
+        const ray = camera.getPickRay(new Cesium.Cartesian2(
+            Math.round(scene.canvas.clientWidth / 2),
+            Math.round(scene.canvas.clientHeight / 2)
+        ));
+
+        const lookAtTarget = scene.globe.pick(ray, scene);
+
+        if (Cesium.defined(lookAtTarget)) {
+            camera.lookAt(lookAtTarget, camera.position);
+        } else {
+            this._log.warn('LookAtTarget cannot be determined. Maybe looking at space.');
+        }
         this._disableMapMoveControls();
```

Comment:
Should _disableMapMoveControls() call be in the if with lookAt()?

Reply:
I thought that in those situation user probably needs to choose "reset to original state" anyway. But i can move disable inside if.

Reply:
In might have shot my self in the leg with using useState in CameraControls3d.jsx because active icon is also set to rotate in these failing cases because mentioned react component does not currently get any response about did operation actually succeeded. Probably would be better to extract state from React component to mapmodule.olcs or separate Handler and implement notify loop where component gets active mode as prop.